Plutarch/Api/V1/AssocMap.hs

```diff
@@ -91,17 +93,20 @@ import qualified Rank2
 
 import Prelude hiding (all, any, filter, lookup, null)
 
-newtype PMap (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
-  deriving (PlutusType, PIsData, PEq, PShow) via (DerivePNewtype (PMap k v) (PBuiltinMap k v))
+data KeyGuarantees = Sorted | Unsorted
+
+newtype PMap (keysort :: KeyGuarantees) (k :: PType) (v :: PType) (s :: S) = PMap (Term s (PBuiltinMap k v))
+  deriving (PlutusType, PIsData, PEq, PShow) via (DerivePNewtype (PMap keysort k v) (PBuiltinMap k v))
+type role PMap nominal nominal representational phantom
```

Comment:
They should all be `nominal`. Anything else is unsound.

Reply:
Do note that we have to do this for all other data types too, and I am in the process of doing that.

Reply:
Okay. I just followed the existing `type role Term phantom representational`, you're saying that's going to change to `type role Term nominal nominal`?

Reply:
Yep. It's broken.